src/System.Runtime.Extensions/src/System/Net/WebUtility.cs

```diff
@@ -50,6 +50,12 @@ public static string HtmlEncode(string value)
             return StringBuilderCache.GetStringAndRelease(sb);
         }
 
+        public static void HtmlEncode(string value, TextWriter output)
+        {
+            // Since we already have a private method that decodes the value using a StringBuilder we don't want to have duplicate code in order to have a better mantainance.
+            output.Write(HtmlEncode(value));
```

Comment:
This is sufficient to bring the functionality to .NET Core, but it can be improved... In desktop, the encoded chars are written directly to the `TextWriter`, whereas with this, a (potentially large) string is being allocated and then that string is written to the `TextWriter`. It'd be nice to avoid the string allocation.

In desktop, the shared implementation is in `HtmlEncode(string, TextWriter)` and `HtmlEncode(string)` allocates a `StringWriter` to pass to `HtmlEncode(string, TextWriter)`. However, in .NET Core, since the other overload wasn't present, `HtmlEncode(string)` was changed to use `StringBuilder` (acquired from a `StringBuilderCache`) instead of using `StringWriter`.

This could be reworked to share an implementation, without having to switch back to the desktop implementation.

Something like...

```c#
public static string HtmlEncode(string value)
{
    ...

    StringBuilder sb = StringBuilderCache.Acquire(value.Length);
    HtmlEncode(value, index, sb, (w, s) => w.Append(s), (w, c) => w.Append(c));
    return StringBuilderCache.GetStringAndRelease(sb);
}

public static void HtmlEncode(string value, TextWriter output)
{
    ...

    HtmlEncode(value, index, output, (w, s) => w.Write(s), (w, c) => w.Write(c));
}

private static unsafe void HtmlEncode<TWriter>(string value, int index, TWriter writer, Action<TWriter, string> writeString, Action<TWriter, char> writeChar)
{
    ...

    // Use writeString(writer, "whatever") or writeChar(writer, 'w')
}
```

Of course, this could be done subsequently in a separate PR.